vpr/src/route/clock_connection_types.cpp

```diff
@@ -0,0 +1,314 @@
+#include "clock_connection_types.h"
+
+#include "globals.h"
+#include "rr_graph2.h"
+
+#include "vtr_assert.h"
+#include "vtr_log.h"
+#include "vtr_error.h"
+
+#include <random>
+#include <math.h>
+
+/*
+ * RoutingToClockConnection (getters)
+ */
+
+ClockConnectionType RoutingToClockConnection::get_connection_type() const {
+    return ClockConnectionType::ROUTING_TO_CLOCK;
+}
+
+/*
+ * RoutingToClockConnection (setters)
+ */
+
+void RoutingToClockConnection::set_clock_name_to_connect_to(std::string clock_name) {
+    clock_to_connect_to = clock_name;
+}
+
+void RoutingToClockConnection::set_clock_switch_name(std::string clock_switch_name) {
+    switch_name = clock_switch_name;
+}
+
+void RoutingToClockConnection::set_switch_location(int x, int y) {
+    switch_location.x = x;
+    switch_location.y = y;
+}
+
+void RoutingToClockConnection::set_switch(int switch_index) {
```

Comment:
Done: commit 71844b54029f7b4f15668726f7c2bfea52c35a4a

Note: I renamed it from switch_index to rr_switch_index instead of adding a comment which I think should make it clear